src/azureStorageExplorer/fileShares/FileShareFS.ts

```
@@ -202,56 +211,30 @@ export class FileShareFS implements vscode.FileSystemProvider {
-            let listFilesAndDirectoriesResult: azureStorage.FileService.ListFilesAndDirectoriesResult = await entry.listFiles(<azureStorage.common.ContinuationToken>undefined!);
-            let entries = listFilesAndDirectoriesResult.entries;
-
-            let directoryResultChild = entries.directories.find(element => element.name === part);
-            if (!!directoryResultChild) {
-                entry = new DirectoryTreeItem(entry, parentPath, directoryResultChild, <azureStorage.FileService.ShareResult>entry.share);
-                parentPath = path.posix.join(parentPath, part);
-            } else {
-                let fileResultChild = entries.files.find(element => element.name === part);
-                if (!!fileResultChild) {
-                    entry = new FileTreeItem(entry, fileResultChild, parentPath, <azureStorage.FileService.ShareResult>entry.share);
-                } else {
-                    throw getFileSystemError(uri, context, vscode.FileSystemError.FileNotFound);
-                }
-            }
-        }
-
-        return entry;
+        throw new RangeError(`Unexpected entry ${entry.constructor.name}.`);
     }
 
-    private async getRoot(uri: vscode.Uri, context: IActionContext): Promise<FileShareTreeItem> {
-        let root = await findRoot(uri, this._fileShareString, context);
-        if (root instanceof FileShareTreeItem) {
-            return root;
+    private async lookup(uri: vscode.Uri | string, context: IActionContext): Promise<EntryTreeItem> {
+        let uriString = uri instanceof vscode.Uri ? uri.path : uri;
+        let ti = await ext.tree.findTreeItem(uriString, context);
+        if (ti instanceof FileShareTreeItem || ti instanceof FileTreeItem || ti instanceof DirectoryTreeItem) {
+            return ti;
         } else {
-            throw new RangeError(`Unexpected entry ${root.constructor.name}.`);
+            throw getFileSystemError(uri, context, vscode.FileSystemError.FileNotFound);
```

This should only happen if ti is undefined. Otherwise you should throw a range error